pkg/broker/broker.go

```diff
@@ -18,45 +18,117 @@ package broker
+	"gopkg.in/yaml.v2"
 	klog "k8s.io/klog/v2"
 )
 
-// NewBroker is a hook that is called with the Options the program is run
+// OverrideChartParams holds optional default values for helm charts
+type OverrideChartParams struct {
+	Mariadb    map[string]interface{} `yaml:"mariadb"`
+	Mongodb    map[string]interface{} `yaml:"mongodb"`
+	Mysql      map[string]interface{} `yaml:"mysql"`
+	Postgresql map[string]interface{} `yaml:"postgresql"`
+	Rabbitmq   map[string]interface{} `yaml:"rabbitmq"`
+	Redis      map[string]interface{} `yaml:"redis"`
+}
+
+// LoadYaml parses param definitions from raw yaml
+func (d *OverrideChartParams) LoadYaml(data []byte) error {
+	err := yaml.UnmarshalStrict(data, d)
+	return err
+}
+
+// ForService returns the parameters for the given service
+func (d *OverrideChartParams) ForService(service string) (map[string]interface{}, bool) {
+	values := map[string]interface{}{}
+
+	switch service {
+	case "mariadb":
+		values = d.Mariadb
+	case "mongodb":
+		values = d.Mongodb
+	case "mysql":
+		values = d.Mysql
+	case "postgresql":
+		values = d.Postgresql
+	case "rabbitmq":
+		values = d.Rabbitmq
+	case "redis":
+		values = d.Redis
+	}
+
+	return values, values != nil
+}
+
+// MinibrokerClient defines the interface of the client the broker operates on
+type MinibrokerClient interface {
+	Init(repoURL string) error
+	ListServices() ([]osb.Service, error)
+	Provision(instanceID, serviceID, planID, namespace string, acceptsIncomplete bool, provisionParams map[string]interface{}) (string, error)
+	Bind(instanceID, serviceID, bindingID string, acceptsIncomplete bool, bindParams map[string]interface{}) (string, error)
+	Unbind(instanceID, bindingID string) error
+	GetBinding(instanceID, bindingID string) (*osb.GetBindingResponse, error)
+	Deprovision(instanceID string, acceptsIncomplete bool) (string, error)
+	LastOperationState(instanceID string, operationKey *osb.OperationKey) (*osb.LastOperationResponse, error)
+	LastBindingOperationState(instanceID, bindingID string) (*osb.LastOperationResponse, error)
+}
+
+// NewBrokerFromOptions is a hook that is called with the Options the program is run
 // with. NewBroker is the place where you will initialize your
 // Broker the parameters passed in.
-func NewBroker(o Options) (*Broker, error) {
+func NewBrokerFromOptions(o Options) (*Broker, error) {
 	klog.V(5).Infof("broker: creating a new broker with options %+v", o)
 	mb := minibroker.NewClient(o.ConfigNamespace, o.ServiceCatalogEnabledOnly)
 	err := mb.Init(o.HelmRepoURL)
 	if err != nil {
 		return nil, err
 	}
 
-	// For example, if your Broker requires a parameter from the command
-	// line, you would unpack it from the Options and set it on the
-	// Broker here.
+	overrideChartParams := &OverrideChartParams{}
+	if len(o.OverrideChartParams) > 0 {
+		data, err := ioutil.ReadFile(o.OverrideChartParams)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to read default chart values file '%q': %w", o.OverrideChartParams, err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to initialize the broker: %w", err)
```

Reply:
Error messages should be lower-case and reflect the current scope, not what the failed function/method does.